core/plugin_api.rs

```diff
@@ -15,9 +15,24 @@ pub use crate::ZeroCopyBuf;
 
 pub type InitFn = fn(&mut dyn Interface);
 
-pub type DispatchOpFn =
-  fn(&mut dyn Interface, &[u8], Option<ZeroCopyBuf>) -> Op;
+pub trait DispatchOpFn:
+  Fn(&mut dyn Interface, &[u8], Option<ZeroCopyBuf>) -> Op
+{
+  fn box_op(self) -> Box<dyn DispatchOpFn>;
+}
+
+impl<D: Fn(&mut dyn Interface, &[u8], Option<ZeroCopyBuf>) -> Op + 'static>
+  DispatchOpFn for D
+{
+  fn box_op(self) -> Box<dyn DispatchOpFn> {
+    Box::new(self)
+  }
+}
```

Comment:
It don't know if this really simplifies things enough to be worth it. It's not much harder to just use `Box::new`. Users will still have to import the trait to use it anyways.